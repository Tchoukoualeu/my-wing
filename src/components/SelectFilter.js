import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import FormControl from '@material-ui/core/FormControl';
import InputLabel from '@material-ui/core/InputLabel';
import Select from '@material-ui/core/Select';
import MenuItem from '@material-ui/core/MenuItem';

const useStyles = makeStyles((theme) => ({
  formControl: {
    margin: theme.spacing(1)
  },
  whiteColor: {
    color: "white"
  }
}));

const SelectFilter = ({ name, label, value, onChangeHandler, items, disabled = false }) => {
  const classes = useStyles();

  return (
    <FormControl className={classes.formControl} disabled={disabled}>
      <InputLabel
        id={`${name}-label`}
        focused={false}
        classes={{root: classes.whiteColor}}
      >
        {label}
      </InputLabel>
      <Select
        labelId={`${name}-label`}
        name={name}
        value={value}
        onChange={onChangeHandler}
        classes={{
          root: classes.whiteColor,
          icon: classes.whiteColor
        }}
      >
        {items.map((item, key) => (
          <MenuItem
            key={key}
            value={item.value}
            disabled={!!item.disabled}
          >
            {item.label}
          </MenuItem>
        ))}
      </Select>
    </FormControl>
  );
};

export default SelectFilter;