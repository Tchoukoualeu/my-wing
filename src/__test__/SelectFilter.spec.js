import React from "react";
import { create } from "react-test-renderer";
import SelectFilter from "../components/SelectFilter";

describe("SelectFilter component", () => {
  const sectionItems = [
    {value: "hot", label: "Hot"},
    {value: "top", label: "Top"},
    {value: "user", label: "User"}
  ];
  const sortItems = [
    {value: "viral", label: "Viral"},
    {value: "top", label: "Top"},
    {value: "time", label: "Time"},
    {value: "rising", label: "Rising"}
  ];
  const windowItems = [
    {value: "day", label: "Day"},
    {value: "week", label: "Week"},
    {value: "month", label: "Month"},
    {value: "year", label: "Year"},
    {value: "all", label: "All"}
  ];

  test("Matches the snapshots for Section", () => {
    sectionItems.forEach(item => {
      const selectFilter = create(
        <SelectFilter
          name="section"
          label="Section"
          value={item.value}
          items={sectionItems}
        />
      );

      expect(selectFilter.toJSON()).toMatchSnapshot();
    });
  });

  test("Matches the snapshot for Sort", () => {
    sortItems.forEach(item => {
      const selectFilter = create(
        <SelectFilter
          name="sort"
          label="Sort"
          value={item.value}
          items={sortItems}
        />
      );

      expect(selectFilter.toJSON()).toMatchSnapshot();
    });
  });

  test("Matches the snapshot for Window", () => {
    windowItems.forEach(item => {
      const selectFilter = create(
        <SelectFilter
          name="window"
          label="Window"
          value={item.value}
          items={windowItems}
        />
      );

      expect(selectFilter.toJSON()).toMatchSnapshot();
    });
  });
});