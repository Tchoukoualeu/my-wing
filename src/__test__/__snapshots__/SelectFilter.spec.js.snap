// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectFilter component Matches the snapshot for Sort 1`] = `
<div
  className="MuiFormControl-root makeStyles-formControl-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root makeStyles-whiteColor-2 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    id="sort-label"
  >
    Sort
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="sort-label mui-component-select-sort"
      className="MuiSelect-root makeStyles-whiteColor-2 MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      id="mui-component-select-sort"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Viral
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput"
      name="sort"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value="viral"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon makeStyles-whiteColor-2"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`SelectFilter component Matches the snapshot for Sort 2`] = `
<div
  className="MuiFormControl-root makeStyles-formControl-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root makeStyles-whiteColor-2 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    id="sort-label"
  >
    Sort
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="sort-label mui-component-select-sort"
      className="MuiSelect-root makeStyles-whiteColor-2 MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      id="mui-component-select-sort"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Top
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput"
      name="sort"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value="top"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon makeStyles-whiteColor-2"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`SelectFilter component Matches the snapshot for Sort 3`] = `
<div
  className="MuiFormControl-root makeStyles-formControl-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root makeStyles-whiteColor-2 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    id="sort-label"
  >
    Sort
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="sort-label mui-component-select-sort"
      className="MuiSelect-root makeStyles-whiteColor-2 MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      id="mui-component-select-sort"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Time
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput"
      name="sort"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value="time"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon makeStyles-whiteColor-2"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`SelectFilter component Matches the snapshot for Sort 4`] = `
<div
  className="MuiFormControl-root makeStyles-formControl-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root makeStyles-whiteColor-2 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    id="sort-label"
  >
    Sort
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="sort-label mui-component-select-sort"
      className="MuiSelect-root makeStyles-whiteColor-2 MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      id="mui-component-select-sort"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Rising
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput"
      name="sort"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value="rising"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon makeStyles-whiteColor-2"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`SelectFilter component Matches the snapshot for Window 1`] = `
<div
  className="MuiFormControl-root makeStyles-formControl-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root makeStyles-whiteColor-2 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    id="window-label"
  >
    Window
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="window-label mui-component-select-window"
      className="MuiSelect-root makeStyles-whiteColor-2 MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      id="mui-component-select-window"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Day
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput"
      name="window"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value="day"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon makeStyles-whiteColor-2"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`SelectFilter component Matches the snapshot for Window 2`] = `
<div
  className="MuiFormControl-root makeStyles-formControl-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root makeStyles-whiteColor-2 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    id="window-label"
  >
    Window
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="window-label mui-component-select-window"
      className="MuiSelect-root makeStyles-whiteColor-2 MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      id="mui-component-select-window"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Week
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput"
      name="window"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value="week"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon makeStyles-whiteColor-2"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`SelectFilter component Matches the snapshot for Window 3`] = `
<div
  className="MuiFormControl-root makeStyles-formControl-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root makeStyles-whiteColor-2 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    id="window-label"
  >
    Window
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="window-label mui-component-select-window"
      className="MuiSelect-root makeStyles-whiteColor-2 MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      id="mui-component-select-window"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Month
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput"
      name="window"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value="month"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon makeStyles-whiteColor-2"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`SelectFilter component Matches the snapshot for Window 4`] = `
<div
  className="MuiFormControl-root makeStyles-formControl-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root makeStyles-whiteColor-2 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    id="window-label"
  >
    Window
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="window-label mui-component-select-window"
      className="MuiSelect-root makeStyles-whiteColor-2 MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      id="mui-component-select-window"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Year
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput"
      name="window"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value="year"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon makeStyles-whiteColor-2"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`SelectFilter component Matches the snapshot for Window 5`] = `
<div
  className="MuiFormControl-root makeStyles-formControl-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root makeStyles-whiteColor-2 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    id="window-label"
  >
    Window
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="window-label mui-component-select-window"
      className="MuiSelect-root makeStyles-whiteColor-2 MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      id="mui-component-select-window"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      All
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput"
      name="window"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value="all"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon makeStyles-whiteColor-2"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`SelectFilter component Matches the snapshots for Section 1`] = `
<div
  className="MuiFormControl-root makeStyles-formControl-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root makeStyles-whiteColor-2 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    id="section-label"
  >
    Section
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="section-label mui-component-select-section"
      className="MuiSelect-root makeStyles-whiteColor-2 MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      id="mui-component-select-section"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Hot
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput"
      name="section"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value="hot"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon makeStyles-whiteColor-2"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`SelectFilter component Matches the snapshots for Section 2`] = `
<div
  className="MuiFormControl-root makeStyles-formControl-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root makeStyles-whiteColor-2 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    id="section-label"
  >
    Section
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="section-label mui-component-select-section"
      className="MuiSelect-root makeStyles-whiteColor-2 MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      id="mui-component-select-section"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Top
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput"
      name="section"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value="top"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon makeStyles-whiteColor-2"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`SelectFilter component Matches the snapshots for Section 3`] = `
<div
  className="MuiFormControl-root makeStyles-formControl-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root makeStyles-whiteColor-2 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    id="section-label"
  >
    Section
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="section-label mui-component-select-section"
      className="MuiSelect-root makeStyles-whiteColor-2 MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      id="mui-component-select-section"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      User
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput"
      name="section"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value="user"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon makeStyles-whiteColor-2"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
